CMakePresets.json

```diff
@@ -0,0 +1,528 @@
+      "configurePreset": "linux-gcc-release-ootree-skeleton-fast",
+      "inherits": "linux-build-base",
+      "configuration": "Release"
+    },
+    {
+      "name": "linux-gcc-release-ootree-skeleton-full",
+      "displayName": "Linux x64: GCC / Release / Out of tree (full)",
+      "configurePreset": "linux-gcc-release-ootree-skeleton-full",
+      "inherits": "linux-build-base",
+      "configuration": "Release"
+    },
+
+
+    {
+      "name": "macos-build-base",
+      "hidden": true,
+      "configuration": "Debug",
+      "condition": {
+        "type": "equals",
+        "lhs": "${hostSystemName}",
+        "rhs": "Darwin"
+      }
+    },
+    {
+      "name": "macos-clang-debug-ootree-skeleton-fast",
+      "displayName": "macOS x64: Clang / Skeleton / Out of tree (fast)",
+      "configurePreset": "macos-clang-debug-ootree-skeleton-fast",
+      "inherits": "macos-build-base"
+    },
+    {
+      "name": "macos-clang-debug-ootree-skeleton-full",
+      "displayName": "macOS x64: Clang / Skeleton / Out of tree (full)",
+      "configurePreset": "macos-clang-debug-ootree-skeleton-full",
+      "inherits": "macos-build-base"
+    },
+    {
+      "name": "macos-clang-debug-intree-skeleton-fast",
+      "displayName": "macOS x64: Clang / Skeleton / In tree (fast)",
+      "configurePreset": "macos-clang-debug-intree-skeleton-fast",
+      "inherits": "macos-build-base"
+    },
+    {
+      "name": "macos-clang-debug-intree-skeleton-full",
+      "displayName": "macOS x64: Clang / Skeleton / In tree (full)",
+      "configurePreset": "macos-clang-debug-intree-skeleton-full",
+      "inherits": "macos-build-base"
+    },
+    {
+      "name": "macos-clang-release-ootree-skeleton-fast",
+      "displayName": "macOS x64: CLang / Release / Out of tree (fast)",
+      "configurePreset": "macos-clang-release-ootree-skeleton-fast",
+      "inherits": "macos-build-base",
+      "configuration": "Release"
+    },
+    {
+      "name": "macos-clang-release-ootree-skeleton-full",
+      "displayName": "macOS x64: CLang / Release / Out of tree (full)",
+      "configurePreset": "macos-clang-release-ootree-skeleton-full",
+      "inherits": "macos-build-base",
+      "configuration": "Release"
+    },
+
+
+    {
+      "name": "windows-build-base",
+      "hidden": true,
+      "configuration": "Debug",
+      "condition": {
+        "type": "equals",
+        "lhs": "${hostSystemName}",
+        "rhs": "Windows"
+      }
+    },
+    {
+      "name": "windows-msvc-debug-ootree-fast",
+      "displayName": "Windows x64: MSVC / Debug / Out of tree (fast)",
+      "configurePreset": "windows-msvc-debug-ootree-fast",
+      "inherits": "windows-build-base"
+    },
+    {
+      "name": "windows-msvc-debug-ootree-full",
+      "displayName": "Windows x64: MSVC / Debug / Out of tree (full)",
+      "configurePreset": "windows-msvc-debug-ootree-full",
+      "inherits": "windows-build-base"
+    },
```

Comment:
I think we can drop this pair (in a separate commit). One debug/release and one intree/ootree is enough. Likewise for macOS and for configuration presets.

Reply:
I don't seem to fully understand how you want to do this since `configurePreset` specifying the name of a configure preset to associate with this build preset. As documentation says:

> If `configurePreset` is not specified, it **must** be inherited from the inherits preset (unless this preset is hidden).

It seems to me we can't use one build preset for two different config presets. But lets investigate this a bit later.

Reply:
No, I mean a very simple thing, not related to the presets and your PR. I noticed that we have the following combinations for macOS/Windows:
- debug ootree
- debug intree
- release ootree

And it seems to me that we should have only:
- debug intree
- release ootree

Reply:
A, got it!